File: Frontend/src/app/DTOs/Challenge/ChallengeOpenDTO.ts
import { Subject } from "src/app/Enums/Subject";

export interface ChallengeOpenDTO {
    challengeId: string;
    name: string;
    createSetName: string;
    subjectMain: Subject;
    subjectSecond: Subject | null;
    numberOfPlayers: number;
    isOwner: boolean;
}

File: Backend/DTOs/Challenge/ChallengeOpenDTO.cs
namespace Learnz.DTOs;
public class ChallengeOpenDTO
{
    public Guid ChallengeId { get; set; }
    public string Name { get; set; }
    public string CreateSetName { get; set; }
    public Subject SubjectMain { get; set; }
    public Subject? SubjectSecond { get; set; }
    public int NumberOfPlayers { get; set; }
    public bool IsOwner { get; set; }
}